pkg/cmd/roachtest/roachtestutil/mixedversion/steps.go

```diff
@@ -283,7 +368,31 @@ func (s resetClusterSettingStep) Run(
 	ctx context.Context, l *logger.Logger, rng *rand.Rand, h *Helper,
 ) error {
 	stmt := fmt.Sprintf("RESET CLUSTER SETTING %s", s.name)
-	return h.ExecWithGateway(rng, nodesRunningAtLeast(s.virtualClusterName, s.minVersion, h), stmt)
+	return serviceByName(h, s.virtualClusterName).ExecWithGateway(
+		rng, nodesRunningAtLeast(s.virtualClusterName, s.minVersion, h), stmt,
+	)
+}
+
+// deleteAllTenantsVersionOverrideStep is a hack that deletes bad data
+// from the `system.tenant_settings` table; specifically an
+// all-tenants (tenant_id = 0) override for the 'version' key. See
+// #125702 for more details.
+//
+// This allows us to continue testing virtual cluster upgrades to
+// versions older than 24.2 without hitting this (already fixed) bug.
```

Comment:
Educational question:

Do we need to do this so *any* `23.2/24.1` tenant can upgrade or just the predecessors that don't contain https://github.com/cockroachdb/cockroach/pull/126676?

If it's the former, then why is the aforementioned commit not enough? Still trying to wrap my head around why we need to do both when it was sufficient to just backport one.

Reply:
You understanding is right: we could do away with this logic if: 1. we wait for that backport to be publicly available on the relevant releases (23.2 and 24.1) and 2. we add a new rule that if `deploymentMode == SharedProcess` then we apply `AlwaysUseLatestPredecessors`.

That would be an ok approach too, but I felt slightly more compelled to do this to see if there were any other version-specific bugs out there. There was a bunch of MT work done during 23.1 and 23.2 so there maybe is a chance there's something to be found by using older releases.

Does that make sense? There's nothing stopping us from removing this logic after a few weeks of no bugs found and using latest predecessors to simplify things.

Reply:
Yeah that makes sense now, I forgot about the backport not being out yet. Thanks for explaining 👍